docs/collection.js

```diff
@@ -427,9 +427,12 @@ class Collection {
      * @param {function(collection, changes)} callback - A function to be called when changes occur.
      *   The callback function is called with two arguments:
      *   - `collection`: the collection instance that changed,
-     *   - `changes`: a dictionary with keys `insertions`, `modifications` and `deletions`,
-     *      each containing a list of indices that were inserted, updated or deleted respectively. If
-     *      partial sync is enabled, an additional key `partial_sync` is added.
+     *   - `changes`: a dictionary with keys `insertions`, `newModifications`, `oldModifications`
+     *      and `deletions`, each containing a list of indices in the collection that were
+     *      inserted, updated or deleted respectively. `deletions` and `oldModifications` are
+     *      indices into the collection before the change happened, while `insertions` and
+     *      `newModifications` are indices into the new version of the collection. If partial sync
```

Comment:
I suggest that we use `query_based_sync` instead of `partial_sync`.

Reply:
Btw, we still have `modifications` so we should document them.

Reply:
Looks like I forgot to clean up a PR back in February. The reference to partial sync can be removed.